cloudify_cli/commands/executions.py

```diff
@@ -484,3 +484,41 @@ def summary(target_field, sub_field, logger, client, tenant_name,
         items,
         'Execution summary by {field}'.format(field=target_field),
     )
+
+
+@executions.command(name='delete',
+                    short_help='Delete finished executions from listing')
+@click.option('-n', '--keep-last', type=click.IntRange(min=1))
+@click.option('-d', '--keep-days', type=click.IntRange(min=1))
+@click.option('-s', '--keep-since', type=click.DateTime(['%Y-%m-%d']))
+@cfy.options.common_options
+@cfy.options.tenant_name(required=False,
+                         resource_name_for_help='executions to delete from')
+@cfy.options.all_tenants
+@cfy.pass_logger
+@cfy.pass_client()
+def delete(keep_last, keep_days, keep_since, logger, client, tenant_name,
+           all_tenants):
+    """Delete executions from the executions list, by specifying a number of
+    executions to keep, a number of days to keep executions for, or a date
+    starting from which to keep executions.
+
+    * Only deletes finished executions, i.e. completed, failed or cancelled.
+
+    * Does not delete the latest deployment environment creation for each
+    deployment.
+
+    Must specify one of the options --keep-last | --keep-days | --keep-since"""
+    utils.explicit_tenant_name_message(tenant_name, logger)
+    if keep_last is None and keep_days is None and keep_since is None:
+        logger.critical('Must specify one of the options --keep-last | '
+                        '--keep-days | --keep-since')
+        return
+    deleted_executions = client.executions.delete(
+        keep_last=keep_last,
+        keep_days=keep_days,
+        keep_since=keep_since,
+        all_tenants=all_tenants,
+    )
+    logger.info('Successfully deleted {0} executions'.format(
```

Comment:
not really important, but a pet peeve:
`logger.info('xxx {0}'.format(x))` <-- does the string interpolation no matter if the log is going to be shown or not, so it always does an allocation and costs performance
`logger.info('xxx %s', x)` <--- the logging framework only does the interpolation when the log is going to be actually shown